client/related_items/ComparisonModal.jsx

```diff
@@ -11,13 +11,37 @@ class ComparisonModal extends React.Component {
     }
 
     let key = 0;
+    let mainFeatures = this.props.mainProduct[1];
+    let comparisonFeatures = this.props.productToCompare[1];
+    let allFeatures = [];
+
+    mainFeatures.forEach(feature => {
+      feature.compValue = '';
+      allFeatures.push(feature);
+    });
+
+    comparisonFeatures.forEach(compFeature => {
+      var index = allFeatures.findIndex(item => item.feature === compFeature.feature);
+      if (index !== -1) {
+        allFeatures[index].compValue = compFeature.value;
+      } else {
+        compFeature.compValue = '';
+        allFeatures.push(compFeature);
+      }
+    });
 
     return (
       <div id='comparisonModal' onClick={this.props.toggleModal}>
         <p>Comparing</p>
-        <h4>{this.props.name}</h4>
-        {this.props.features.map(feature =>
-          <div key={`${key++}`}>{feature.feature}: {feature.value}</div>
+        <h4 class='modal-col-1'>{this.props.mainProduct[0]}</h4>
+        <h4 class='modal-col-3'>{this.props.productToCompare[0]}</h4>
+
+        {allFeatures.map(feature =>
+          <div>
+            <div class='modal-col-1'>{feature.value}</div>
+            <div class='modal-col-2'>{feature.feature}</div>
+            <div class='modal-col-3'>{feature.compValue}</div>
+          </div>
```

Comment:
This looks really good to me, Jorge. Do you think it might be better to create helper functions to implement your allFeatures array? It seems like there's a lot going on in your render method and it would be easier to understand/better separation of concerns if you had a descriptively-titled function like 'createProductComparison'. You might also consider making this a functional-stateless component (because it's not using state at all).

Reply:
Definitely! I'm going to take a pass soon to clean up my code - I was very much in a "just get it to work" mode, thanks for pointing it out. And yes I can probably make this stateless, I frankly can't remember why I made it stateful 🤦